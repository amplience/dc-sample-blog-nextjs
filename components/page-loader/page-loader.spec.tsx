/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import PageLoader from './page-loader';
import { shallow } from 'enzyme';

describe('PageLoader', () => {
  test('renders page loader', async () => {
    const component = await renderer.create(<PageLoader />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('prevents on click events when loaded', () => {
    const mockPreventDefault = jest.fn();
    const mockStopPropagation = jest.fn();
    const wrapper = shallow(<PageLoader />);
    wrapper
      .find('div.page-loading')
      .simulate('click', { preventDefault: mockPreventDefault, stopPropagation: mockStopPropagation });

    expect(mockPreventDefault).toHaveBeenCalled();
    expect(mockStopPropagation).toHaveBeenCalled();
  });
});
