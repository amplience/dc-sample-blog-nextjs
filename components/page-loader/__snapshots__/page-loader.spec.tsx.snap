// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLoader renders page loader 1`] = `
<div
  className="jsx-1481206946 page-loading"
  onClick={[Function]}
>
  <div
    className="jsx-1481206946 spinner"
  />
</div>
`;
